File: Algorithms/452. Minimum Number of Arrows to Burst Balloons/javascript/solution.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function (points) {
    if (points.length === 0) return 0;

    points.sort((a, b) => a[1] - b[1]);

    let arrows = 1,
        firstEnd = points[0][1];
    for (let [xStart, xEnd] of points) {
        if (xStart > firstEnd) {
            arrows++;
            firstEnd = xEnd;
        }
    }
    return arrows;
};

File: Algorithms/452. Minimum Number of Arrows to Burst Balloons/kotlin/solution.kt
class Solution {
    fun findMinArrowShots(points: Array<IntArray>): Int {
        if (points.isEmpty()) return 0

        points.sortBy({ it[1] })

        var arrows = 1
        var firstEnd = points[0][1]
        for (i in 1..points.size - 1) {
            if (points[i][0] > firstEnd) {
                arrows++
                firstEnd = points[i][1]
            }
        }
        return arrows
    }
}
